README.adoc

```diff
@@ -28,15 +28,132 @@ Synposis
 
-        . *tasks*: This directory includes tasks for installing and configuring Rudder.
+        . *roles/ansible-deploy-rudder/tasks*: This directory includes tasks for installing and configuring Rudder.
 
-        . *vars*: This directory includes all variables necessary for the installation of Rudder (account creation, etc.)
+        . *roles/ansible-deploy-rudder/vars*: This directory includes all variables necessary for the installation of Rudder (account creation, etc.)
+        
+        . *roles/ansible-deploy-rudder/defaults*: Contains variables used in Rudder installation tasks.
+
+        . *roles/ansible-deploy-rudder/templates*: Contains Jinja2 templates (.j2).
+
+        . *roles/ansible-deploy-rudder/meta* Contains metadata used to deploy the role in Ansible Galaxy.
 
-        . *defaults*: Contains variables used in Rudder installation tasks.
+=== Modules
 
-        . *meta* Contains metadata used to deploy the role in Ansible Galaxy.
+This section aims to give and explain the different parameters possible in each module.
+
+==== Module: 'rudder_server_settings'
+
+This module is used to set up Rudder via API calls.
+
+.Table 'rudder_server_settings' module options
+[cols="1,1,1,3"] 
+|===
+|Name | Type |Required|Description
+
+|rudder_url
+|*false*
+|str
+|Providing Rudder server IP address. Defaults to localhost of the target node if not set, with certificate validation disabled, unless explicitly enabled by setting validate_certs.
+
+|rudder_token
+|*false*
+|str
+|Providing Rudder server token. Defaults to the content of /var/rudder/run/api-token if not set.
+
+|name
+|*true*
+|str
+|The name of the parameter to set.
+
+|value
+|*true*
+|str
+|The value defined to modify a given parameter name.
+
+|validate_certs
+|*false*
+|boolean
+|Choosing either to ignore or not Rudder certificate validation. Defaults to *true*.
+|===
+
+Example of use:
+
+----
+     rudder_server_settings:
+       rudder_url: "https://localhost/rudder"
+       name: "enable_self_validation"
+       value: "false"
+       validate_certs: False
+----
+
+==== Module: 'rudder_node_settings'
+
+This module allows you to configure the settings of the nodes in Rudder.
+
+.Table 'rudder_server_settings' module options
+[cols="1,1,1,3"] 
+|===
+|Name | Type |Required|Description
+
+|rudder_url
+|*false*
+|str
+|Providing Rudder server IP address. Defaults to localhost of the target node if not set, with certificate validation disabled, unless explicitly enabled by setting validate_certs.
+
+|rudder_token
+|*false*
+|str
+|Providing Rudder server token. Defaults to the content of /var/rudder/run/api-token if not set.
+
+|validate_certs
+|*false*
+|boolean
+|Choosing either to ignore or not Rudder certificate validation. Defaults to true.
```

Comment:
i'm unsure of what it should do - using True or False returns the same result. I would expect that either one would fail as I have self-signed certificates

Reply:
This is now fixed. Commit: https://github.com/Normation/rudder-ansible/pull/6/commits/4ad465ac9c3f41c3d1f1b9999ed8c162225d7297